Framework/Beamline/inc/MantidBeamline/Beamline.h

```diff
@@ -0,0 +1,72 @@
+#ifndef MANTID_BEAMLINE_BEAMLINE_H_
+#define MANTID_BEAMLINE_BEAMLINE_H_
+
+#include "MantidBeamline/DllConfig.h"
+#include <boost/shared_ptr.hpp>
+
+namespace Mantid {
+namespace Beamline {
+
+class ComponentInfo;
+class DetectorInfo;
+/** Beamline : Also known as Instrument 2.0
+
+  This is the top-level object for accessing ComponentInfo, DetectorInfo and
+  modern beamline functionality.
+
+  Since ComponentInfo and DetectorInfo have non-owning pointers to each other,
+  this type ensures
+  that client code is not exposed to creation or setup internals. Beamline owns
+  both ComponentInfo and DetectorInfo.
+
+  Beamline is deliberately trivial to copy.
+
+  Copyright &copy; 2017 ISIS Rutherford Appleton Laboratory, NScD Oak Ridge
+  National Laboratory & European Spallation Source
+
+  This file is part of Mantid.
+
+  Mantid is free software; you can redistribute it and/or modify
+  it under the terms of the GNU General Public License as published by
+  the Free Software Foundation; either version 3 of the License, or
+  (at your option) any later version.
+
+  Mantid is distributed in the hope that it will be useful,
+  but WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+  GNU General Public License for more details.
+
+  You should have received a copy of the GNU General Public License
+  along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+  File change history is stored at: <https://github.com/mantidproject/mantid>
+  Code Documentation is available at: <http://doxygen.mantidproject.org>
+*/
+class MANTID_BEAMLINE_DLL Beamline {
```

Comment:
There was a previous chat on the general slack thread, and I think Duc suggested "Beamline" a name. Instrument 2.0 doesn't seem to cut, it and `Instrument` is too easily confused with existing `Instrument 1.0` IMHO. I do agree that `Beamline` itself might not be descriptive enough. Open for suggestions... maybe

* `Geometry`  - as in `Beamline::Geometry`, too easily confused with namespace `Geometry`?
* `BeamlineInstrument`?